### ISSUE TITLE
parentType.getFields is not a function

### ISSUE THREAD

**Issue by kassim** [apollographql/apollo-codegen#141]
On the latest version of codegen (via the apollo-android project) I'm getting the following stack trace

```
:app:generateDevDebugApolloIR
TypeError: parentType.getFields is not a function
    at Compiler.resolveFields (/Users/kassim/git/android-v2/app/build/apollo-codegen/node_modules/apollo-codegen/src/compilation.js:305:35)
    at Compiler.resolveFields (/Users/kassim/git/android-v2/app/build/apollo-codegen/node_modules/apollo-codegen/src/compilation.js:330:67)
    at Compiler.resolveFields (/Users/kassim/git/android-v2/app/build/apollo-codegen/node_modules/apollo-codegen/src/compilation.js:330:67)
    at Compiler.compileFragment (/Users/kassim/git/android-v2/app/build/apollo-codegen/node_modules/apollo-codegen/src/compilation.js:162:63)
    at Compiler.resolveFields (/Users/kassim/git/android-v2/app/build/apollo-codegen/node_modules/apollo-codegen/src/compilation.js:353:79)
    at Compiler.resolveFields (/Users/kassim/git/android-v2/app/build/apollo-codegen/node_modules/apollo-codegen/src/compilation.js:330:67)
    at Compiler.compileFragment (/Users/kassim/git/android-v2/app/build/apollo-codegen/node_modules/apollo-codegen/src/compilation.js:162:63)
    at Compiler.resolveFields (/Users/kassim/git/android-v2/app/build/apollo-codegen/node_modules/apollo-codegen/src/compilation.js:353:79)
    at /Users/kassim/git/android-v2/app/build/apollo-codegen/node_modules/apollo-codegen/src/compilation.js:365:48
    at Array.map (native)
```

See another error reported to apollo-android too https://github.com/apollographql/apollo-android/issues/521

I'll try to share recreation steps later.

**Comment by chrisvis** [comment]
I'm experiencing the same issue. When debugging through the code, I noticed it happens when the `parentType` parameter in `collectFields` is of type `GraphQLUnionType`

**Comment by martijnwalraven** [comment]
@chrisvis: Ah, thanks for tracking this down!

**Comment by martijnwalraven** [comment]
Should be fixed in v0.12.9.

**Comment by chrisvis** [comment]
Wow, that's fast!
